### PR TITLE
feat: add table sort in url and enable for full page tables

### PR DESCRIPTION
**Pull request by aaron-steinfeld**

### CODE REVIEW

File: projects/components/src/table/table.component.ts [hunk 1]
```diff
@@ -149,13 +151,16 @@ export class TableComponent
...
     AfterViewInit,
+    OnDestroy,
     ColumnConfigProvider,
     TableDataSourceProvider,
     FilterProvider,
     ColumnStateChangeProvider,
     RowStateChangeProvider {
   private static readonly PAGE_INDEX_URL_PARAM: string = 'page';
   private static readonly PAGE_SIZE_URL_PARAM: string = 'page-size';
+  private static readonly SORT_COLUMN_URL_PARAM: string = 'sort';
```
Comment (anandtiwary) [inline projects/components/src/table/table.component.ts:162]: sort-attribute? or sort-on ?

File: projects/components/src/table/table.component.ts [hunk 2]
```diff
@@ -453,12 +471,15 @@ export class TableComponent
     }
   }
 
-  private initializeUrl(): void {
-    if (this.paginator) {
-      this.onPageChange({
```
Comment (anandtiwary) [inline projects/components/src/table/table.component.ts:458]: How is this being done now?

File: projects/components/src/table/table.component.ts (hunk 1 above)
Comment (aaron-steinfeld) [reply projects/components/src/table/table.component.ts]: updated to `sort-by` based on offline conversation

File: projects/components/src/table/table.component.ts (hunk 2 above)
Comment (aaron-steinfeld) [reply projects/components/src/table/table.component.ts:458]: It didn't feel valuable to put the default page info in the url on load. I can buy that argument for something like the selected tab, but for page size I think the simpler url for the more common use case is better.

Comment (anandtiwary) [reply projects/components/src/table/table.component.ts:458]: Okay I agree with that.

**Pull request merged**